frame/session/src/lib.rs

```diff
@@ -350,97 +351,134 @@ impl<AId> SessionHandler<AId> for TestSessionHandler {
 	}
 }
 
-pub trait Config: frame_system::Config {
-	/// The overarching event type.
-	type Event: From<Event> + Into<<Self as frame_system::Config>::Event>;
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
 
-	/// A stable ID for a validator.
-	type ValidatorId: Member + Parameter;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
 
-	/// A conversion from account ID to validator ID.
-	///
-	/// Its cost must be at most one storage read.
-	type ValidatorIdOf: Convert<Self::AccountId, Option<Self::ValidatorId>>;
+	/// The pallet's config trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event> + IsType<<Self as frame_system::Config>::Event>;
 
-	/// Indicator for when to end the session.
-	type ShouldEndSession: ShouldEndSession<Self::BlockNumber>;
+		/// A stable ID for a validator.
+		type ValidatorId: Member + Parameter + MaybeSerializeDeserialize;
 
-	/// Something that can predict the next session rotation. This should typically come from the
-	/// same logical unit that provides [`ShouldEndSession`], yet, it gives a best effort estimate.
-	/// It is helpful to implement [`EstimateNextNewSession`].
-	type NextSessionRotation: EstimateNextSessionRotation<Self::BlockNumber>;
+		/// A conversion from account ID to validator ID.
+		///
+		/// Its cost must be at most one storage read.
+		type ValidatorIdOf: Convert<Self::AccountId, Option<Self::ValidatorId>>;
 
-	/// Handler for managing new session.
-	type SessionManager: SessionManager<Self::ValidatorId>;
+		/// Indicator for when to end the session.
+		type ShouldEndSession: ShouldEndSession<Self::BlockNumber>;
 
-	/// Handler when a session has changed.
-	type SessionHandler: SessionHandler<Self::ValidatorId>;
+		/// Something that can predict the next session rotation. This should typically come from the
+		/// same logical unit that provides [`ShouldEndSession`], yet, it gives a best effort estimate.
+		/// It is helpful to implement [`EstimateNextNewSession`].
+		type NextSessionRotation: EstimateNextSessionRotation<Self::BlockNumber>;
 
-	/// The keys.
-	type Keys: OpaqueKeys + Member + Parameter + Default;
+		/// Handler for managing new session.
+		type SessionManager: SessionManager<Self::ValidatorId>;
 
-	/// The fraction of validators set that is safe to be disabled.
-	///
-	/// After the threshold is reached `disabled` method starts to return true,
-	/// which in combination with `pallet_staking` forces a new era.
-	type DisabledValidatorsThreshold: Get<Perbill>;
+		/// Handler when a session has changed.
+		type SessionHandler: SessionHandler<Self::ValidatorId>;
 
-	/// Weight information for extrinsics in this pallet.
-	type WeightInfo: WeightInfo;
-}
+		/// The keys.
+		type Keys: OpaqueKeys + Member + Parameter + Default + MaybeSerializeDeserialize;
```

Comment:
What's the reason behind the extra `MaybeSerializeDeserialize` trait bound here?

Reply:
Since `Keys` is used in storage, otherwise throws error 
```
 the trait `Deserialize<'_>` is not implemented for `<T as pallet::Config>::Keys`
note: required because of the requirements on the impl of `Deserialize<'_>` for `std::vec::Vec<(<T as frame_system::Config>::AccountId, <T as pallet::Config>::ValidatorId, <T as pallet::Config>::Keys)>`
    = note: required by `next_value`
```